lib/Sema/CSSimplify.cpp

```diff
@@ -1139,9 +1139,29 @@ ConstraintSystem::matchFunctionTypes(FunctionType *func1, FunctionType *func2,
   // that expect a function that takes a single tuple (of the same
   // arity).
   auto isSingleParam = [&](ArrayRef<AnyFunctionType::Param> params) {
-    return (params.size() == 1 &&
-            params[0].getLabel().empty() &&
-            !params[0].isVariadic());
+    if (params.size() != 1)
+      return false;
+
+    const auto &param = params.front();
+    if (param.isVariadic() || param.isInOut())
+      return false;
+
+    auto paramType = param.getPlainType();
+    // Support following case which was allowed until 5:
+    // ``swift
```

Comment:
This isn't markdown, I would suggest removing the back ticks :)

Reply:
Sure :)